state/protocol/protocol_state/epochs/fallback_statemachine_test.go

```diff
@@ -663,3 +674,66 @@ func (s *EpochFallbackStateMachineSuite) TestEpochFallbackStateMachineInjectsMul
 			"final view should be greater than final view of test")
 	}
 }
+
+// TestProcessingMultipleEventsAtSameBlock tests that the state machine can process multiple events at the same block.
+// EpochFallbackStateMachineSuite has to be able to process any combination of events in any order at the same block.
+// This test generates a random amount of setup, commit and recover events and processes them in random order.
+func (s *EpochFallbackStateMachineSuite) TestProcesingMultipleEventsAtTheSameBlock() {
```

Comment:
could you please add ejection events here too? I think we have uncovered edge cases, where we should reject a recovery event that would be re-emitting an ejected node. Though, I think the current implementation would accept this edge case. 

You could add this as a ToDo to  https://github.com/onflow/flow-go/issues/6019. It would be more or less a test verifying that the current implementation has problems. My gut feeling is that the use if `partentState` causes the problems (not 100 sure that there is really an uncovered edge case through)

Reply:
I have added some stuff in [`6e09f32` (#6079)](https://github.com/onflow/flow-go/pull/6079/commits/6e09f324a93d35e40b329e0f2d779f940cadc301) but it will require extra love when implementing 6019. But you are right, currently there is an edge case and epoch recover event is accepted even though it readmits an ejected entity. We actually end up in broken state where current epoch has an entity ejected and in next epoch it's not.

Reply:
👍 added [todo](https://github.com/onflow/flow-go/issues/6019#issuecomment-2189663711) to [issue 6019](https://github.com/onflow/flow-go/issues/6019)